### PR TITLE
[W5.7][W17-1]Anubhav

### PR DESCRIPTION
**Pull request by anubh-v**

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
-import java.util.HashSet;
-import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.Arrays;
+import seedu.addressbook.commands.ThenListCommand;
```
Comment (ongspxm) [inline src/seedu/addressbook/parser/Parser.java]: should this be put on the following block instead? considering its importing from `seedu`

Comment (anubh-v) [reply src/seedu/addressbook/parser/Parser.java]: Yes, that's right. Thanks for the notice.

File: test/expected.txt [hunk 2]
```diff
@@ -31,7 +31,8 @@
 || Example: exit
 || ===================================================
 || Enter command: || [Command entered:  delete 1]
-|| The person index provided is invalid
+||
```
Comment (amrut-prabhu) [inline test/expected.txt:34]: Good job for updating the I/O tests!

File: src/seedu/addressbook/commands/ThenListCommand.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.AddressBook;
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/ThenListCommand.java:3]: ~Remember to follow the same order of imports as the rest of the codebase!~
Edit: Not required till AB4

Might be easier if you update the IntelliJ settings, so you won't have to worry about manually changing it.

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -3,14 +3,15 @@
...
 import static seedu.addressbook.common.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.Set;
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/parser/Parser.java:8]: Same as before. Follow the standard order of imports.

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -49,7 +49,7 @@ Help is also shown if you enter an incorrect command e.g. `abcd`
 
 == Adding a person: `add`
 
-Adds a person to the address book. +
```
Comment (amrut-prabhu) [inline docs/UserGuide.adoc:52]: Good job for updating the User Guide!

File: src/seedu/addressbook/commands/ThenListCommand.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
+public class ThenListCommand extends Command {
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/ThenListCommand.java:8]: Something to think about: Is `ThenListCommand` an actual command? Can the user explicitly execute it? This leads to the question: can you use inheritance here instead?
Can `ThenListCommand` be an abstract class or an interface?
`AddCommand` and `DeleteCommand` can be thought of as `ThenListCommand`s.

File: src/seedu/addressbook/commands/ThenListCommand.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        ListCommand listCmd = new ListCommand();
+        listCmd.setData(this.prevCommand.addressBook, this.prevCommand.relevantPersons);
+
+        return listCmd.execute();
+
+    }
+
+    @Override
+    public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> relevantPersons) {
+        prevCommand.addressBook = addressBook;
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/ThenListCommand.java:33]: Would it be better to use setters here instead?

File: src/seedu/addressbook/commands/ThenListCommand.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public CommandResult execute() {
+        this.prevCommand.execute();
+
+        ListCommand listCmd = new ListCommand();
+        listCmd.setData(this.prevCommand.addressBook, this.prevCommand.relevantPersons);
+
+        return listCmd.execute();
+
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/ThenListCommand.java:28]: Unnecessary whitespace?

**Review by amrut-prabhu — APPROVED** [review]
Good work on the enhancement @anubh-v 
Try to update the JUnit tests as well next time so that you can check whether there are any regressions.
Please close the PR after reading the comments.

**Comment by ongspxm** [comment]
Lolz

without wax, ongspxm
